src/prometheus_alert_rules/memory.rules

```diff
@@ -0,0 +1,69 @@
+groups:
+- name: HostMemory
+  rules:
+  - record: node_memory_MemUsed_percentage
+    expr: |
+      100 * (1 - (
+        (
+          node_memory_MemFree_bytes
+          + node_memory_Cached_bytes
+          + node_memory_Buffers_bytes
+          + node_memory_SReclaimable_bytes
+        ) / node_memory_MemTotal_bytes
+      ))
+  - record: node_memory_SwapUsed_percentage
+    expr: |
+      100 * (1 - (
+        (
+          node_memory_SwapFree_bytes
+          + node_memory_SwapCached_bytes
+        ) / node_memory_SwapTotal_bytes
+      ))
+  - alert: HostMemoryFillsUp
+    expr: predict_linear(node_memory_MemUsed_percentage[30m], 5*60) >= 90
```

Comment:
based on my experience, on servers with a lot of memory (500GB+). 90% is perfectly normal memory usage, especially in environments that make use of hugepages, DPDK etc. like anything telco-related, does it make any sense to set this threshold so low? or is there any way to make it configurable? in the legacy stack (nrpe charm) things like thresholds are exposed in the charm config

Reply:
If that is the case, then we need to set the threshold higher. We won't be adding config options to set these as dynamic/configurable rules are expected to be injected using [cos-config](https://github.com/canonical/cos-configuration-k8s-operator)

Reply:
95% would be ok for these? I think that servers like that are more corner cases in which it maybe ok to have even 98% usage.

Would be possible to overwrite these alert rules w/ [cos-config](https://github.com/canonical/cos-configuration-k8s-operator) or in such situation these alerts rules needs to be silenced and new needs to provided w/ [cos-config](https://github.com/canonical/cos-configuration-k8s-operator)?

Reply:
If there won't be any config options in this charm and we're forced to manually tune everything via cos-config then we can leave it at 90%.